src/System.Management.Automation/utils/tracing/PSSysLogProvider.cs

```diff
@@ -0,0 +1,337 @@
+#if UNIX
+//
+//    Copyright (C) Microsoft.  All rights reserved.
+//
+using System.Diagnostics.Eventing;
+using System.Management.Automation.Internal;
+using System.Text;
+using System.Collections.Generic;
+
+namespace System.Management.Automation.Tracing
+{
+    /// <summary>
+    /// SysLog LogProvider implementation
+    /// </summary>
+    internal class PSSysLogProvider : LogProvider
+    {
+        private static SysLogProvider s_provider;
+
+        // by default, do not include analytic events
+        internal const PSKeyword DefaultKeywords = (PSKeyword) (0xFFFFFFFFFFFFFFFF & ~(ulong)PSKeyword.UseAlwaysAnalytic);
+
+        /// <summary>
+        /// Class constructor
+        /// </summary>
+        static PSSysLogProvider()
+        {
+            s_provider = new SysLogProvider(ConfigPropertyAccessor.Instance.GetSysLogIdentity(),
+                                            ConfigPropertyAccessor.Instance.GetLogLevel(),
+                                            ConfigPropertyAccessor.Instance.GetLogKeywords(),
+                                            ConfigPropertyAccessor.Instance.GetLogChannels());
+        }
+
+        /// <summary>
+        /// Defines a thread local StringBuilder for building event payload strings.
+        /// </summary>
+        /// <remarks>
+        /// NOTE: do not access this field directly, use the PayloadBuilder
+        /// property to ensure correct thread initialization; otherwise, a null reference can occur.
+        /// </remarks>
+        [ThreadStatic]
+        private static StringBuilder _payloadBuilder;
+
+        private static StringBuilder PayloadBuilder
+        {
+            get
+            {
+                if (_payloadBuilder == null)
+                {
+                    // First time initialization for this thread.
```

Comment:
Please remove the obvious comment or clarify why we use 200.

Reply:
I'll clarify the comment because it's a common bug when using ThreadStatic; it must be explicitly initialized for each thread.